Sources/GoodExtensions/UIKit/UIViewExtensions.swift

```diff
@@ -316,3 +334,110 @@ public extension GRActive where Base: UIView {
 
 }
 
+// MARK: - Dimming
+
+public extension GRActive where Base: UIView {
+
+    private final class DimView: UIView {
+
+        init(intensity alpha: CGFloat = 0.3) {
+            super.init(frame: .zero)
+
+            translatesAutoresizingMaskIntoConstraints = false
+            backgroundColor = .black.withAlphaComponent(alpha)
+        }
+
+        required init?(coder: NSCoder) {
+            fatalError("init(coder:) has not been implemented")
+        }
+
+        func changeIntensity(_ alpha: CGFloat) {
+            backgroundColor = .black.withAlphaComponent(alpha)
+        }
+
+    }
+
+    func dim(intensity alpha: CGFloat = 0.3, duration: CGFloat = 0.3) {
+        undim()
+        let dimView = DimView(intensity: 0)
+
+        addSubview(dimView)
+        NSLayoutConstraint.activate([
+            dimView.topAnchor.constraint(equalTo: base.topAnchor),
+            dimView.leadingAnchor.constraint(equalTo: base.leadingAnchor),
+            dimView.trailingAnchor.constraint(equalTo: base.trailingAnchor),
+            dimView.bottomAnchor.constraint(equalTo: base.bottomAnchor)
+        ])
+
+        UIView.animate(withDuration: duration) {
+            dimView.changeIntensity(alpha)
+        }
+    }
+
+    func undim(duration: CGFloat = 0.3) {
+        let dimView = base.firstSubview(forType: DimView.self)
+        UIView.animate(
+            withDuration: duration,
+            animations: {
+                dimView?.changeIntensity(0)
+            },
+            completion: { [weak base] _ in
+                dimView?.removeFromSuperview()
+                base?.layoutSubviews()
+            }
+        )
+    }
+
+}
+
+// MARK: - Blurring
+
+extension UIView {
+
+    func createBlurryImage(_ blurRadius: Double) -> UIImage? {
+        UIGraphicsBeginImageContext(bounds.size)
+        guard let currentContext = UIGraphicsGetCurrentContext() else {
+            return nil
+        }
+        layer.render(in: currentContext)
+        guard let image = UIGraphicsGetImageFromCurrentImageContext(),
+            let blurFilter = CIFilter(name: "CIGaussianBlur") else {
+            UIGraphicsEndImageContext()
+            return nil
+        }
+        UIGraphicsEndImageContext()
+
+        blurFilter.setDefaults()
+
+        blurFilter.setValue(CIImage(image: image), forKey: kCIInputImageKey)
+        blurFilter.setValue(blurRadius, forKey: kCIInputRadiusKey)
+
+        var convertedImage: UIImage?
+        let context = CIContext(options: nil)
+        if let blurOutputImage = blurFilter.outputImage,
+            let cgImage = context.createCGImage(blurOutputImage, from: blurOutputImage.extent) {
+            convertedImage = UIImage(cgImage: cgImage)
+        }
+
+        return convertedImage
+    }
+
+}
+
+// MARK: - Subview search
+
+public extension GRActive where Base: UIView {
+
+    func firstSubview<T>(forType type: T.Type) -> T? {
+        for subview in base.subviews where subview is T {
```

Comment:
Tu by som revertol obidva tie `for` cykly, vacsinou chceme asi hladat view, ktory sme pridali my, teda bude pravdepodobne niekde na konci hierarchie. Zbytocne tak nebudeme prechadzat do hlbky views, ktore tam pridal system.

Reply:
Ja by som to nechal. Môžeme to do budúcna použiť na iný scenár ako odstraňovanie dimu a bluru. V týchto dvoch pripravovať máš ale pravdu. Väčšinou vieš na aký view si to pridal, takze nieje potrebné to volat